File: Spark-Learning/src/java/com/atguigu/spark/core/transformatioins/Union.java
package com.atguigu.spark.core.transformatioins;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.VoidFunction;

import java.util.Arrays;

public class Union {
    public static void main(String[] args) {
        SparkConf conf = new SparkConf();
        conf.setMaster("local");
        conf.setAppName("test");

        JavaSparkContext sc = new JavaSparkContext(conf);
        sc.setLogLevel("Error");

        JavaRDD<Integer> rdd1 = sc.parallelize(Arrays.asList(1, 2, 3, 4, 5));
        JavaRDD<Integer> rdd2 = sc.parallelize(Arrays.asList(6, 7, 8, 9, 10));
        JavaRDD<Integer> result = rdd1.union(rdd2);
        result.foreach(new VoidFunction<Integer>() {
            @Override
            public void call(Integer integer) throws Exception {
                System.out.println(integer);
            }
        });

    }
}

File: Spark-Learning/src/scala/com/atguigu/sparkcore/transformation/Union.scala
package com.atguigu.sparkcore.transformation

import org.apache.spark.rdd.RDD
import org.apache.spark.{SparkConf, SparkContext}

/**
  * 合并两个数据集。两个数据集的类型要一致。
  * 返回新的RDD的分区数是合并RDD分区数的总和。
  */
object Union {
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf()
    conf.setMaster("local")
    conf.setAppName("test")
    val sc = new SparkContext(conf)
    val rdd1 = sc.parallelize(List[String]("a", "b", "c"), 2)
    val rdd2 = sc.parallelize(List[String]("d", "e", "f"), 3)
    println(s"rdd1 partitions length = ${rdd1.getNumPartitions}")
    println(s"rdd2 partitions length = ${rdd2.getNumPartitions}")

    val unit: RDD[String] = rdd1.union(rdd2)
    println(s"unit partitions length = ${unit.getNumPartitions}")
    unit.foreach(println)
  }
}
